### ISSUE TITLE
Detect custom resources with more than one name_property

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#372]
### What category of cop is this?:

Please try to select as few as possible.

- [ ] Deprecation - A cop that alerts to a deprecation in the Chef Infra Client
- [ ] Correctness - A cop that alerts when incorrect coding behavior is being used
- [ ] Sharing - A cop that alerts to missing functionality necessary for sharing cookboooks with others
- [ ] Style - A cop that alerts to a style best practice
- [ ] Simplification - A cop that alerts to ways code can be simplified through refactoring
- [ ] Effortless - A cop that alerts on code that must be resolved to move to the Effortless pattern
- [ ] Policyfiles - A cop that alerts on code that must be resolved to move to the Policyfiles pattern

### Describe the new cop:
You can't have more than one name_property. We should detect this and alert

### What it would trigger on:

```ruby
property :product_name, String, name_property: true
property :resource_name, String, name_property: true
```

### What it would autocorrect to if applicable
N/A

**Comment by tas50** [comment]
I think there's actually a valid use for this now that I think about it. This is weird, but it would work just fine and someone is probably using it in a totally valid way.
